### PR TITLE
[FOLLOW-UP][TT-XLA][PyTorch] Align SST2 Dataset with New Infra and JAX Implementation

### PR DESCRIPTION
**Pull request by mmilosevicTT**
### Ticket
This is a follow-up of https://github.com/tenstorrent/tt-blacksmith/issues/232

### Problem description
Since new infra was added in https://github.com/tenstorrent/tt-blacksmith/pull/263, our `SSTDataset` is not aligned to it. 

### What's changed
This PR aligns `SSTDataset` to newly added infra. Amongst changes there are:
- Switch of `SSTDataset` to new infra
- Addition of `split` field to `BaseDataset`, to store split of the dataset
- Addition of `collate_fn` to `BaseDataset`, to allow for custom collator
- Fix of dataset size logging in Gemma, Qwen and Llama experiments
- Fix of validation if no tokenizer was provided for Gemma and Llama
- Renaming of `llama` directory in `datasets` to `sst`
- Move of `collate_fn_for_causal_lm` to `tools`
- Addition of `shuffle` and `drop_last` to Qwen, Llama and Gemma experiemnts
- Switch `stanfordnlp/sst2` to `glue/sst2` to align with JAX implementation

### Checklist
As this PR is chaning different experiments, we need to confirm all of them are still working
- [x] Testing of Gemma experiment
- [x] Testing of Qwen experiment
- [x] Testing of FFE Llama experiment
- [x] Testing of XLA Llama experiment
- [x] Testing of JAX implementation for LoRA/DoRA for LLama

### CODE REVIEW

File: blacksmith/datasets/torch/sst2/sst2_dataset.py [hunk 1]
```diff
@@ -54,8 +54,8 @@ def _tokenize_function(self, example):
         return example
 
     def _prepare_dataset(self):
-        print(f"Loading dataset ({self.config.dataset_id})...")
-        raw_dataset = load_dataset(self.config.dataset_id, split=self.split)
+        print(f"Loading dataset ({self.config.dataset_id}/{self.config.dataset_configuration})...")
```
Comment (pmarkovicTT) [inline blacksmith/datasets/torch/sst2/sst2_dataset.py]: I would remove print since we should use logger and we log when the dataset is loaded in the training script. If we notice later that we need to log in this, or any other step in the dataset, we can add logger as well, now it seems to me it might be overkill.
